src/ui/common/src/components/pages/metric/id/index.tsx

```diff
@@ -0,0 +1,462 @@
+  //   {};
+  // console.log('logs: ', logs);
+  // const operatorError =
+  //   workflow?.operatorResults[metricOperatorId]?.result?.exec_state?.error;
+  // console.log('operatorError', operatorError);
+
+  // // Get the execution state of the operator:
+  // // TODO: Show execution state in a badge component next to title of operator.
+  // const execState: ExecState =
+  //   workflow?.operatorResults[metricOperatorId]?.result?.exec_state;
+  // console.log('execState: ', execState);
+
+  const metricResult = workflow.operatorResults[metricOperatorId];
+
+  useEffect(() => {
+    // TODO: Update this to contain the name of the operator
+    document.title = 'Metric Details | Aqueduct';
+    dispatch(handleGetWorkflow({ apiKey: user.apiKey, workflowId }));
+  }, []);
+
+  useEffect(() => {
+    if (workflow.dagResults.length > 0) {
+      dispatch(
+        handleGetOperatorResults({
+          apiKey: user.apiKey,
+          workflowDagResultId,
+          operatorId: metricOperatorId,
+        })
+      );
+    }
+  }, [workflow.dagResults]);
+
+  // Set up different metric input types for rendering in the inputs list.
+  // TODO: transform/handle response from API and render these appropriately.
+  const mockMetricInputs = [
+    {
+      name: 'bool_artifact',
+      nodeType: NodeType.BoolArtifact,
+      nodeIcon: boolArtifactNodeIcon,
+    },
+    {
+      name: 'checkOperator',
+      nodeType: NodeType.CheckOp,
+      nodeIcon: checkOperatorNodeIcon,
+    },
+    {
+      name: 'database_node',
+      nodeType: NodeType.LoadOp,
+      nodeIcon: databaseNodeIcon,
+    },
+    {
+      name: 'function_operator_node',
+      nodeType: NodeType.FunctionOp,
+      nodeIcon: functionOperatorNodeIcon,
+    },
+    {
+      name: 'generic_artifact_node',
+      nodeType: NodeType.GenericArtifact,
+      nodeIcon: genericArtifactNodeIcon,
+    },
+    {
+      name: 'image_artifact_node',
+      nodeType: NodeType.ImageArtifact,
+      nodeIcon: imageArtifactNodeIcon,
+    },
+    {
+      name: 'jsonArtifactNode',
+      nodeType: NodeType.JsonArtifact,
+      nodeIcon: jsonArtifactNodeIcon,
+    },
+    {
+      name: 'metric_operator_node',
+      nodeType: NodeType.MetricOp,
+      nodeIcon: metricOperatorNodeIcon,
+    },
+    {
+      name: 'numeric_artifact_node',
+      nodeType: NodeType.NumericArtifact,
+      nodeIcon: numericArtifactNodeIcon,
+    },
+    {
+      name: 'string_artifact_node',
+      nodeType: NodeType.StringArtifact,
+      nodeIcon: stringArtifactNodeIcon,
+    },
+    {
+      name: 'table_artifact_node_icon',
+      nodeType: NodeType.TableArtifact,
+      nodeIcon: tableArtifactNodeIcon,
+    },
+  ];
+
+  const artifactTypeToIconMapping = {
+    [ArtifactType.String]: stringArtifactNodeIcon,
+    [ArtifactType.Bool]: boolArtifactNodeIcon,
+    [ArtifactType.Numeric]: numericArtifactNodeIcon,
+    [ArtifactType.Dict]: dictArtifactNodeIcon,
+    // TODO: figure out if we should use other icon for tuple
```

Comment:
@vsreekanti @cw75 Do you have any thoughts on what font awesome icons we should use for these artifact types?

Reply:
I think @cw75 already did this for the DAG visualization in the `Node` implementation. Maybe we could copy those over?